File: leetcodes/1381/solution.py
class CustomStack:

    def __init__(self, maxSize: int):
        self.life = []
        self.maxSize = maxSize

    def push(self, x: int) -> None:
        if len(self.life) < self.maxSize:
            self.life.append(x)

    def pop(self) -> int:
        return self.life.pop() if self.life else -1

    def increment(self, k: int, val: int) -> None:
        # Increment the first min(k, len(self.life)) elements
        limit = min(k, len(self.life))
        for i in range(limit):
            self.life[i] += val

File: leetcodes/1381/solution.cpp
#include <algorithm>
#include <cstdint>
#include <vector>

class CustomStack {
 public:
  CustomStack(int maxSize)
      : max_size(maxSize), stack(std::vector(max_size, -1)), current_size(-1) {}

  void push(int x) {
    if (stack[max_size - 1] != -1) {
      return;
    }
    current_size += 1;
    stack[current_size] = x;
  }

  int pop() {
    if (current_size == -1) {
      return -1;
    }
    int ans = stack[current_size];
    stack[current_size] = -1;
    current_size -= 1;
    return ans;
  }

  void increment(int k, int val) {
    for (int i = 0; i < std::min(max_size, k); i++) {
      if (stack[i] != -1)
        stack[i] += val;
    }
  }

 private:
  std::int32_t max_size;
  std::vector<int32_t> stack;
  std::int32_t current_size;
};
